processor/transformer/transformer.go

```diff
@@ -342,6 +353,12 @@ func (trans *handle) transform(
 		// Transform is one to many mapping so returned
 		// response for each is an array. We flatten it out
 		for _, transformerResponse := range batch {
+			// flip back sourceID and original sourceID
+			if transformerResponse.Metadata.OriginalSourceID != "" {
```

Comment:
Not sure if we should do this, there's a possibility that this data undergoes updates during the transformation, which could impact its reliability for our purposes.

Reply:
I agree.
The same pattern is being used for `messageID` in several other places in processor too.
Maybe that's something we should take a look at too.

Reply:
Just found out metadata can't be updated in a user transformation, so should be a non-issue.